addons/website_links/tests/test_link_tracker.py

```diff
@@ -0,0 +1,93 @@
+from odoo.tests import TransactionCase, tagged
+
+
+@tagged('post_install', '-at_install')
+class TestLinkTracker(TransactionCase):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+
+        cls.company_1 = cls.env['res.company'].create({
+            'name': 'Company 1',
+        })
+        cls.company_2 = cls.env['res.company'].create({
+            'name': 'Company 2',
+        })
+
+        cls.website_1 = cls.env['website'].create({
+            'name': 'Website 1',
+            'domain': 'https://company1.example.com',
+            'company_id': cls.company_1.id,
+        })
+        cls.website_2 = cls.env['website'].create({
+            'name': 'Website 2',
+            'domain': 'https://company2.example.com',
+            'company_id': cls.company_2.id,
+        })
+
+        cls.company_1.write({'website_id': cls.website_1.id})
+        cls.company_2.write({'website_id': cls.website_2.id})
+
+        cls.test_user = cls.env['res.users'].create({
+            'name': 'Test User',
+            'login': 'test_user',
+            'company_ids': [(6, 0, [cls.company_1.id, cls.company_2.id])],
+            'company_id': cls.company_1.id,
+        })
+
+    def setUp(self):
+        super().setUp()
+        self.env.user.company_id = self.company_1
+
+    def test_compute_short_url_host(self):
```

Comment:
Those tests are a bit hard to follow... can't you write a clear input / output structure so that we understand each case ? Notably usage of test_user in one test and not the another is not crystal clear to me.